### PR TITLE
Mygaru Id System: Initial release

### PR DESCRIPTION
**Pull request by GeneGenie**
## Type of change

- [x] New id system

## Description of change
Add new RTD adapter MyGaru

- contact email of the adapter’s maintainer: prebid@mygaru.com
- test parameters for validating bids: 

```javascript
pbjs.setConfig({
    userSync: {
        auctionDelay: 100,
        userIds: [{
            name: 'mygaruId',
        }]
    }
});
```

## Other information
- [docs pr](https://github.com/prebid/prebid.github.io/pull/5040)

### CODE REVIEW

**Comment by GeneGenie** [comment]
@ChrisHuie hi, is this scheme better than RTD scheme ? =)

**Comment by GeneGenie** [comment]
@ncolletti hi, can you get this?

File: package-lock.json
Comment (ncolletti) [inline package-lock.json:1]: remove package-lock changes

File: test/spec/modules/mygaruIdSyestem_spec.js
Comment (ncolletti) [inline test/spec/modules/mygaruIdSyestem_spec.js:1]: typo on this file name

File: modules/mygaruSystem.md
Comment (ncolletti) [inline modules/mygaruSystem.md:1]: fix typo, should be `mygaruIdSystem`

File: test/spec/modules/mygaruIdSyestem_spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import { mygaruIdSubmodule } from 'modules/mygaruIdSystem.js';
+import { server } from '../../mocks/xhr';
+
+describe('MygaruID module', function () {
+  it('should respond with async callback and get valid id', async () => {
```
Comment (ncolletti) [inline test/spec/modules/mygaruIdSyestem_spec.js:5]: please add additional coverage tests for when your server responds with errors or empty responses. As well, what is the expected id value data type and length? in your docs it is closer to: `vatfS0UEgXShEqYuQy0lpzWl7rSuKeNs0U8C8841KoYbFUiyvTpS`

File: test/spec/modules/mygaruIdSyestem_spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const callBackSpy = sinon.spy();
+    const expectedUrl = `https://ident.mygaru.com/v2/id?gdprApplies=0`;
```
Comment (ncolletti) [inline test/spec/modules/mygaruIdSyestem_spec.js:7]: What is the behavior when gdpr Applies, is a gvlid needed to operate in EU?

File: modules/mygaruIdSystem.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      {
+        method: 'GET',
+        contentType: 'application/json'
+      }
+    );
+  });
+}
+
+/** @type {Submodule} */
+export const mygaruIdSubmodule = {
```
Comment (ncolletti) [inline modules/mygaruIdSystem.js:48]: please add updates to the following files, match existing id module patterns:
modules/.submodules.json
modules/userId/eids.js
modules/userId/eids.md
modules/userId/userId.md

File: modules/mygaruIdSystem.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+function requestRemoteIdAsync(url) {
+  return new Promise((resolve) => {
+    ajax(
+      url,
+      {
+        success: response => {
+          const jsonResponse = JSON.parse(response);
+          const { iuid } = jsonResponse;
+          resolve(iuid);
```
[the remark this answers is not in the document]
Comment (GeneGenie) [reply modules/mygaruIdSystem.js:32]: is this validation required ?  in case there is no `iuid`, it should still resolve with `undefined` or `null` ?

File: modules/mygaruIdSystem.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * get the MyGaru Id from local storages and initiate a new user sync
+   * @function
+   * @param {SubmoduleConfig} [config]
+   * @param {ConsentData} [consentData]
+   * @returns {{id: string | undefined}}
+   */
+  getId(config, consentData) {
+    const gdprApplies = consentData && typeof consentData.gdprApplies === 'boolean' && consentData.gdprApplies ? 1 : 0;
+    const gdprConsentString = gdprApplies ? consentData.consentString : undefined;
```
[the remark this answers is not in the document]
Comment (GeneGenie) [reply modules/mygaruIdSystem.js:71]: Good question, what is the best practice here? 
i mean, it doesn't store anything... 
and for now won't operate in gdpr zone. Do you recommend to delete this ?

File: test/spec/modules/mygaruIdSyestem_spec.js (hunk 1 above)
Comment (GeneGenie) [reply test/spec/modules/mygaruIdSyestem_spec.js:5]: 84 symbols +-

File: modules/mygaruIdSystem.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (GeneGenie) [reply modules/mygaruIdSystem.js:71]: I clarified with backend: currently, when gdpr applies from EU, then request is ignored and fake id is returned

File: modules/mygaruIdSystem.js (hunk 3 above)
Comment (GeneGenie) [reply modules/mygaruIdSystem.js:48]: modules/userId/eids.js  doesn't contain module specific values anymore

File: package-lock.json
Comment (ncolletti) [reply package-lock.json:1]: @GeneGenie these changes still need to be removed

File: modules/mygaruIdSystem.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ncolletti) [reply modules/mygaruIdSystem.js]: this is fine to leave as-is, other id modules check for existence before resolving, the user id module will do the validation for a truthy response

File: modules/mygaruIdSystem.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (ncolletti) [reply modules/mygaruIdSystem.js:75]: thanks, please include an update to your docs pr verifying your ID module does not operate in EU.

**Review by ncolletti — APPROVED** [review]
looks good, please make the update to your docs pr for GDPR
